crates/oxc_isolated_declarations/src/global_symbol_binding_tracker.rs

```diff
@@ -0,0 +1,176 @@
+use std::cell::Cell;
+
+use oxc_allocator::{Allocator, Vec};
+#[allow(clippy::wildcard_imports)]
+use oxc_ast::ast::*;
+use oxc_ast::AstBuilder;
+#[allow(clippy::wildcard_imports)]
+use oxc_ast::{visit::walk::*, Visit};
+use oxc_span::{Atom, GetSpan, Span};
+use oxc_syntax::scope::{ScopeFlags, ScopeId};
+use rustc_hash::FxHashSet;
+
+/// Declaration scope.
+#[derive(Debug)]
+struct Scope {
+    has_symbol_binding: bool,
+    has_global_this_binding: bool,
+}
+
+impl Scope {
+    fn new() -> Self {
+        Self { has_symbol_binding: false, has_global_this_binding: false }
+    }
+}
+
+/// Linear tree of declaration scopes.
+pub struct GlobalSymbolBindingTracker<'a> {
+    levels: Vec<'a, Scope>,
+    computed_properties_using_non_global_symbol: FxHashSet<Span>,
+    computed_properties_using_non_global_global_this: FxHashSet<Span>,
```

Comment:
I think in theory we could just store the `start` instead of the whole `Span` and still be correct, but my guess is that it wouldn't make much difference in performance.